### DISCUSSION TITLE
Vuepress 2 - migrate v1.x custom component that uses resolveSidebarItems

### DISCUSSION THREAD

**Discussion by vtt-lair** [vuepress/core#1549] [Q&A]
We have a custom component created by previous developer that uses resolveSidebarItems from v1.x to create links on a parent/first ReadMe. This then shows the links in a ul and allows you to navigate to those pages.

This is the current code:
```
<template>
    <div>
        <ul class="link-list">
            <li v-for="link in links"><a :href="link.path">{{ link.title }}</a></li>
        </ul>
    </div>
</template>

<script>
import { resolveSidebarItems } from '@vuepress/theme-default/util';
const BASE = '/documentation/internal-documentation';

export default {
    props: ['displayText'],
    computed: {
        sidebarItems () {
            return resolveSidebarItems(
                this.$page,
                this.$page.regularPath,
                this.$site,
                this.$localePath
            )
        },
        links () {
            return this.sidebarItems
                .filter((item) => item.path !== this.$page.regularPath)
                .map((item) => {
                    return {
                        title: item.title,
                        path: item.path ? `${BASE}${item.path}` : `${BASE}${item.children[0].path}`
                    }
                }
            );
        }
    }
}
</script>
```

I can't figure out of Vuepress 2 has the same functionality. I did find a resolveSidebarItems function under @vuepress/theme-default, under the lib/client/composables/useSidebarItems.js but couldn't figure out if:

1. This has the same functionality as the one in v1.x.
2. How to actually import it.

Not sure if I'm being an idiot here, whether the I should just scrap the component and try to figure something else out.

Getting the following error:
![image](https://github.com/vuepress/core/assets/33215552/7c811883-28fe-4623-9aed-0064cc10e4d1)

**Comment by Mister-Hope** [comment]
You probably need to rewrite every component to get it migrate to VuePress2, we rebuilt our core structure to increase performance.

Try digging through v2 theme code could be a option
